src/main/java/org/example/data/DataSource.java

```diff
@@ -0,0 +1,20 @@
+package org.example.data;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+
+public class DataSource {
+
+    private static final String DB_URL = "jdbc:h2:mem:testdb;DB_CLOSE_DELAY=-1";
+    private static final String DB_USER = "sa";
+    private static final String DB_PASSWORD = "";
```

Comment:
과제의 측면에서는 그냥 이 위치에 DB 접속 정보를 노출해도 전혀 상관 없고, 귀찮게 굳이 환경변수까지 쓸 정도는 아닌 것 같기는 하지만, 다른 부분에서 리뷰할 만한 부분이 별로 생각나지 않아 이곳에 리뷰 남깁니다!

DB 접속정보는 환경변수로 따로 빼서 아래와 같은 방식으로 실행 시 받아오는 것도 괜찮지 않을까 싶어요!

```suggestion
    private static final String DB_URL = getEnv("DB_URL");
    private static final String DB_USER = getEnv("DB_USER");
    private static final String DB_PASSWORD = getEnv("DB_PASSWORD");
```

Reply:
`DB_URL=jdbc:h2:mem:testdb\;DB_CLOSE_DELAY=-1\;;DB_USER=sa;DB_PASSWORD=;`

Reply:
@1-vL 
오~ getEnv는 그럼 어디에 선언해서 사용하나요?
ConfigurationProperties 비슷한 방식으로 관리하는걸까요?

Reply:
보통 말씀하신 대로 ConfigurationProperties와 유사하게 .env 같은 파일에 저런 환경변수를 저장해서 사용합니다!
보통은 Spring으로 @Value("${jwt.secret}")와 같은 식으로 쓰긴했어요!

그냥 System.getenv(key); 하는 식으로는 현재 실행 중인 환경의 환경 변수를 그대로 가져와 JAVA_HOME 같은거 보는데 사용할 수 있겠지만, 인텔리제이 같은 IDE에서 .env 파일을 사용해서 넘겨주거나, 아니면 도커 이미지를 만들 때 환경변수로 넘겨줄 값을 넣어주면 그렇게도 사용할 수 있다고 알고 있습니다!

Reply:
앗 System 클래스의 getenv 말씀하시는거였군용

아마 그 매서드 deprecated 됐을거에요!!

Reply:
실제로 DB 서버가 띄워지는 상황이면 data source 정보의 노출을 방어할 수 있어야 한다는 것 백번 동의합니다!! 사실 그런 상황이면 이미 spring을 쓰고 있을 것 같긴 합니다 😄 yml을 parsing해서 동적으로 적용하는 것을 직접 구현해보는 것도 재미있을지도...? 🤔 (이번에 한다는 뜻 절대 아닙니다 ㅎ)

Reply:
솔직히 재미없을것같습니다.

Reply:
ㅋㅋㅋㅋㅋㅋㅋㅋ ㅇㅈ

Reply:
정말 다 맞는 말씀인 것 같습니다! (재미는 둘째치고...)
실제 상황이면 당연히 더 나은 기술을 쓰고 있을 거 같아요.
이상하게도 System 클래스의 getenv가 다시 deprecated에서 돌아오기는 한 것 같더라구요. (잘은 모르겠지만...) 😕
(물론 OS 환경변수 가져오는 만큼 사용은 웬만하면 지양하라지만 지금은 너무 손쉬운 해결책 같았습니다.) 😔